controllers/argocd/status.go

```diff
@@ -250,16 +252,33 @@ func (r *ReconcileArgoCD) reconcileStatusNotifications(cr *argoprojv1a1.ArgoCD)
 func (r *ReconcileArgoCD) reconcileStatusHost(cr *argoprojv1a1.ArgoCD) error {
 	cr.Status.Host = ""
 	cr.Status.Phase = "Available"
-	if cr.Spec.Server.Route.Enabled {
-		if !IsRouteAPIAvailable() {
-			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
-			return nil
-		}
+
+	if (cr.Spec.Server.Route.Enabled || cr.Spec.Server.Ingress.Enabled) && IsRouteAPIAvailable() {
```

Comment:
I think this logic to adjust the status based on OpenShift route host should go to gitops operator's ReconcilerHook.

Reply:
@wtam2018 All the OpenShift Routes logic should move out of this operator. 

This will be a major change which we should target in the near future. I think it is out of the scope of this PR.